### PR TITLE
Add bool to JWT to indicate whether group owned executors should be used.

### PR DESCRIPTION
**Pull request by vadimberezniker**
I decided against introducing yet another cache for group information
for the time being. We can revisit if we run into other group
information that we need to access frequently.

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/auth/auth.go [hunk 1]
```diff
@@ -17,10 +16,12 @@ import (
 	"github.com/buildbuddy-io/buildbuddy/server/tables"
 	"github.com/buildbuddy-io/buildbuddy/server/util/capabilities"
 	"github.com/buildbuddy-io/buildbuddy/server/util/db"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
 	"github.com/buildbuddy-io/buildbuddy/server/util/lru"
 	"github.com/buildbuddy-io/buildbuddy/server/util/random"
 	"github.com/buildbuddy-io/buildbuddy/server/util/status"
 
+	"github.com/coreos/go-oidc"
```
Comment (tylerwilliams) [inline enterprise/server/auth/auth.go:24]: this makes it unclear where oidc comes from (go-oidc has package "oidc") which is why I aliased it initially

File: enterprise/server/auth/auth.go [hunk 2]
```diff
@@ -467,25 +459,27 @@ func (a *OpenIDAuthenticator) lookupAPIKeyGroupFromAPIKey(apiKey string) (interf
...
+			GroupID:                akg.GetGroupID(),
+			AllowedGroups:          []string{akg.GetGroupID()},
+			Capabilities:           capabilities.FromInt(akg.GetCapabilities()),
+			UseGroupOwnedExecutors: akg.GetUseGroupOwnedExecutors(),
+		}
+		return assembleJWT(ctx, claims)
 	}
 
-	return assembleJWT(ctx, userID, groupID, allowedGroups, capabilities)
+	return "", status.FailedPreconditionErrorf("No user/group to generate JWT for")
```
Comment (tylerwilliams) [inline enterprise/server/auth/auth.go:482]: nice

File: server/testutil/testauth/testauth.go [hunk 3]
```diff
@@ -44,10 +44,11 @@ var (
...
 	jwt.StandardClaims
-	UserID        string                   `json:"user_id"`
-	GroupID       string                   `json:"group_id"`
-	AllowedGroups []string                 `json:"allowed_groups"`
-	Capabilities  []akpb.ApiKey_Capability `json:"capabilities"`
+	UserID                 string                   `json:"user_id"`
+	GroupID                string                   `json:"group_id"`
+	AllowedGroups          []string                 `json:"allowed_groups"`
+	Capabilities           []akpb.ApiKey_Capability `json:"capabilities"`
+	UseGroupOwnedExecutors bool                     `json:"use_group_owned_executors,omitempty"`
```
Comment (tylerwilliams) [inline server/testutil/testauth/testauth.go:51]: just checking: should we have omitempty on other fields? Why this field?

**Review by tylerwilliams — APPROVED** [review]
LGTM (few nits)

File: enterprise/server/auth/auth.go (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/server/auth/auth.go]: Done.

**Pull request merged**